### PR TITLE
Feature/135919 rsvp edd tpp contact subscription

### PR DESCRIPTION
**Pull request by jesseeproductions**
_Ref:_ [C#135919](https://central.tri.be/issues/135919)

### CODE REVIEW

File: src/Tribe/Admin/Notices.php [hunk 1]
```diff
@@ -56,80 +38,156 @@ protected function slug( $string ) {
...
+			return;
+		}
+
+		// Bail if previously dismissed this notice.
+		if ( Tribe__Admin__Notices::instance()->has_user_dimissed( $this->slug( 'missing-application-credentials' ) ) ) {
+			return;
+		}
+
+		// Bail if the Application Credentials are Saved.
+		$options       = tribe( 'tickets.hubspot' )->get_all_options();
```
Comment (sc0ttkclark) [inline src/Tribe/Admin/Notices.php]:
```suggestion
		$options = tribe( 'tickets.hubspot' )->get_all_options();

```

File: src/Tribe/Admin/Notices.php [hunk 2]
```diff
@@ -56,80 +38,157 @@ protected function slug( $string ) {
...
-		$missing_credentials = tribe( 'tickets.hubspot.api' )->has_required_fields();
+	public function render_missing_application_credentials_notice( $settings ) {
+
+		// Get link to APIs Tab.
+		$url = $settings->get_url( [
+			'page' => 'tribe-common',
+			'tab'  => 'addons',
+		] );
+
+		$message = sprintf( '<div><p>%s, <a href="%s" target="_blank">%s</a>.</p></div>',
```
Comment (sc0ttkclark) [inline src/Tribe/Admin/Notices.php]:
```suggestion
		$message = sprintf( '<div><p>%1$s, <a href="%2$s" target="_blank">%3$s</a>.</p></div>',
```

File: src/Tribe/Admin/Notices.php [hunk 3]
```diff
@@ -56,80 +38,157 @@ protected function slug( $string ) {
...
+		// Get link to APIs Tab.
+		$url = $settings->get_url( [
+			'page' => 'tribe-common',
+			'tab'  => 'addons',
+		] );
 
-		return ! empty( $transient ) && empty( $access_token );
-	}
 
+		$message = sprintf( '<div><p>%s, <a href="%s" target="_blank">%s</a>.</p></div>',
```
Comment (sc0ttkclark) [inline src/Tribe/Admin/Notices.php]:
```suggestion
		$message = sprintf( '<div><p>%1$s, <a href="%2$s" target="_blank">%3$s</a>.</p></div>',
```

File: src/Tribe/Properties/Event_Data.php [hunk 4]
```diff
@@ -40,20 +40,17 @@ public function get_event_values( $prefix, $post_id ) {
...
 			[
 				'property' => $prefix . 'event_cost',
-				'value'    => $event->cost,
+				'value'    => html_entity_decode( $event->cost ),
 			],
-			//todo must be midnight utc in milliseconds sending as 1576800000000 by hubspot uses 1576800000
 			[
 				'property' => $prefix . 'event_start_datetime_utc',
-				'value'    => date( 'U', strtotime( 'midnight', ( \Tribe__Date_Utils::wp_strtotime( $event->start_date_utc ) ) ) ) * 1000,
+				'value'    => ms_timestamp( strtotime( 'midnight', ( \Tribe__Date_Utils::wp_strtotime( $event->start_date_utc ) ) ) ),
```
Comment (sc0ttkclark) [inline src/Tribe/Properties/Event_Data.php]: Freaky function!

Comment (jesseeproductions) [reply src/Tribe/Properties/Event_Data.php]: Yeah, it is a fun one.

**Pull request merged**